tests/unit_tests/test_util.py

```diff
@@ -818,3 +818,16 @@ def test_json_dump_uncompressed_with_numpy_datatypes():
 )
 def test_are_windows_paths_on_same_drive(path1, path2, expected):
     assert util.are_paths_on_same_drive(path1, path2) == expected
+
+
+def test_lazy_import_adds_attribute_to_parent_module():
```

Comment:
You could write a more realistic test by launching a Python subprocess that performs imports and does checks. This would go under "functional tests" (the name is not descriptive; it's just tests that primarily launch subprocesses); here's an example script: tests/system_tests/test_functional/wb_logging/last_resort.py and its corresponding test tests/system_tests/test_functional/wb_logging/test_wb_logging_last_resort.py

I'm not sure whether this test is really complete otherwise; I would have to look up the details of how Python imports interact with `sys.modules` to know whether there is an uncaught edge case.